src/plugins/firebasePlugin.ts

```diff
@@ -1,11 +1,21 @@
+import type { App } from 'firebase-admin/app';
 import { initializeApp, applicationDefault } from 'firebase-admin/app';
 import { getAuth } from 'firebase-admin/auth';
 
-initializeApp({
-  credential: applicationDefault(),
-});
+// eslint-disable-next-line functional/no-let
+let firebaseApp: App | undefined;
+
+const initApp = () => {
+  if (!firebaseApp) {
+    firebaseApp = initializeApp({
+      credential: applicationDefault(),
+    });
+  }
+};
 
 export const createUserInFirebase = async (email: string, displayName: string) => {
+  initApp();
```

Comment:
had to lazy load in order to be able to mock on tests. otherwise module with real firebase-admin/auth would load before being able to mock and call to firebase servers were being made.  spent some time trying to learn how to mock dependencies completely using mocha/chai/sinon but without success so far. so I opted for this approach to unblock me

Reply:
Good addition

Reply:
This makes sense to me @fredericobraga85 👍